optuna/study.py

```diff
@@ -306,12 +306,27 @@ def trials_dataframe(self, include_internal_fields=False):
 
             Get an objective value and a value of parameter ``x`` in the first row.
 
-            >>> df = study.trials_dataframe()
-            >>> df
-            >>> df.value[0]
-            0.0
-            >>> df.params.x[0]
-            1.0
+            .. testsetup:: *
+
+                import optuna
+
+                def objective(trial):
```

Comment:
How about moving `objective` to the  `doctest` block? I prefer an example's being self-contained unless it's too noisy (e.g., too many imports).

Reply:
Moved everything to a single `.. testcode::` (which in contrast to `.. doctest::` does check the outputs) with a more realistic scenario. It became a lot more comprehensible not hiding parts of the code as you suggested.

Reply:
Improved it slightly.
<img width="696" alt="Screen Shot 2019-11-25 at 18 21 42" src="https://user-images.githubusercontent.com/5983694/69527844-77652700-0fb0-11ea-8976-d78b073c4f25.png">